pkg/scheduler/backend/workloadmanager/workloadmanager.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workloadmanager
+
+import (
+	"fmt"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	fwk "k8s.io/kube-scheduler/framework"
+)
+
+// WorkloadManager is the central source of truth for the state of pods belonging to Workload objects.
+// It is designed to be driven explicitly by the scheduler's event handlers to ensure thread safety
+// and avoid race conditions with the main scheduling queue.
+type WorkloadManager interface {
```

Comment:
nit: let's add a comment that the current implementtion assumes that pod.WorkloadReference is immutable.
If we ever decide to change that, the proper updatePod will need to be implemented.